src/olympia/addons/models.py

```diff
@@ -237,57 +238,137 @@ def not_disabled_by_mozilla(self):
+            elif theme_review:
+                qs = qs.exclude(reviewerflags__needs_admin_theme_review=True)
             else:
                 qs = qs.exclude(reviewerflags__needs_admin_code_review=True)
         return qs
 
+    def get_listed_pending_manual_approval_queue(
+        self,
+        admin_reviewer=False,
+        recommendable=False,
+        statuses=amo.VALID_ADDON_STATUSES,
+        types=amo.GROUP_TYPE_ADDON,
+    ):
+        if types not in (amo.GROUP_TYPE_ADDON, amo.GROUP_TYPE_THEME):
+            raise ImproperlyConfigured(
+                'types needs to be either GROUP_TYPE_ADDON or GROUP_TYPE_THEME'
+            )
+        theme_review = types == amo.GROUP_TYPE_THEME
+        qs = self.get_base_queryset_for_queue(
+            admin_reviewer=admin_reviewer,
+            # The select related needed to avoid extra queries in other queues
+            # typically depend on current_version, but we don't care here, as
+            # it's a pre-review queue. We'll make the select_related() calls we
+            # need ourselves.
+            select_related_fields_for_listed=False,
+            # We'll filter on pending_rejection below without limiting
+            # ourselves to the current_version.
+            exclude_listed_pending_rejection=False,
+            theme_review=theme_review,
+        )
+        filters = (
+            Q(
+                status__in=statuses,
+                type__in=types,
+                versions__channel=amo.RELEASE_CHANNEL_LISTED,
+                versions__files__status=amo.STATUS_AWAITING_REVIEW,
+                versions__reviewerflags__pending_rejection=None,
+            )
+            & ~Q(disabled_by_user=True)
+        )
+        if recommendable:
+            filters &= Q(promotedaddon__group_id=RECOMMENDED.id)
+        elif not theme_review:
+            filters &= ~Q(promotedaddon__group_id=RECOMMENDED.id) & (
+                Q(versions__files__is_webextension=False)
+                | Q(reviewerflags__auto_approval_disabled=True)
+                | Q(reviewerflags__auto_approval_disabled_until_next_approval=True)
+                | Q(reviewerflags__auto_approval_delayed_until__gt=datetime.now())
+                | Q(
+                    promotedaddon__group_id__in=(
+                        group.id for group in PRE_REVIEW_GROUPS
+                    )
+                )
+            )
+
+        return (
+            # We passed select_related_fields_for_listed=False but there are
+            # still base select_related() fields in that queryset that are
+            # applied in all cases that we don't want.
+            qs.select_related(None)
+            .filter(filters)
+            .select_related('reviewerflags')
+            .annotate(
+                first_version_nominated=Min('versions__nomination'),
+                latest_version=Func(F('versions__version'), function='LOWER'),
```

Comment:
why we do we need to apply the `LOWER` function?

Reply:
It's a trick I should document in a comment. 

- If you just do `annotate(latest_version=F('versions__version'))` with no other annotations, everything is fine.
- If you just do `annotate(first_version_nominated=Min('versions__nomination'))`, then Django detects you want a `GROUP BY addon.id`, everything is fine.
- If you combine them with `.annotate(latest_version=F('versions__version'), first_version_nominated=Min('versions__nomination'))`, then Django detects you want a `GROUP BY addon.id`, but also adds the other field to it, causing a `GROUP BY addons.id, versions.version` which we don't want at all!
- Adding the `Func()` prevents that (Django can no longer detect what to add to the `GROUP BY`, as you're not annotating with a field directly), but you can't have an empty `Func()`, so I used `LOWER` as it's harmless here.

Reply:
4175920ae1076a4d47f1f70b7f899a8359892ebd adds some comments and also a test for the query being generated